docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,532 @@
+* `schema`: the structure that defines the schema format
+
+The schema is identical to the :ref:`schema file defined for Java client applications<schema_registry_tutorial_definition>`.
+Notice in the output above, the schema is escaped JSON, i.e., the double quotes are preceded with backslashes.
+
+Based on the schema id, you can also retrieve the associated schema by querying |sr| REST endpoint:
+
+.. sourcecode:: bash
+
+   $ curl --silent -X GET http://localhost:8081/schemas/ids/1 | jq .
+   {
+     "schema": "{\"type\":\"record\",\"name\":\"Payment\",\"namespace\":\"io.confluent.examples.clients.basicavro\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"amount\",\"type\":\"double\"}]}"
+   }
+
+If you are using |c3|, you can view the topic schema easily from the UI:
+
+.. figure:: c3-schema-transactions.png
+    :align: center
+
+
+
+Schema IDs in Messages
+^^^^^^^^^^^^^^^^^^^^^^
+
+Integration with |sr-long| means that Kafka messages do not need to be written with the entire Avro schema.
+Instead, Kafka messages are written with the schema id.
+The producers writing the messages and the consumers reading the messages must be using the same |sr| to get the same mapping between a schema and schema id.
+
+In this example, a producer sends the new schema for `Payments` to |sr|.
+|sr| registers this schema `Payments` to the subject `transactions-value`, and returns the schema id of `1` to the producer.
+The producer caches this mapping between the schema and schema id for subsequent message writes, so it only contacts |sr| on the first schema write.
+When a consumer reads this data, it sees the Avro schema id of `1` and sends a schema request to |sr|.
+|sr| retrieves the schema associated to schema id `1`, and returns the schema to the consumer.
+The consumer caches this mapping between the schema and schema id for subsequent message reads, so it only contacts |sr| the on first schema id read.
+
+
+Auto Schema Registration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, client applications automatically register new schemas.
+If they produce new messages to a new topic, then they will automatically try to register new schemas.
+This is very convenient in development environments, but in production environments we recommend that client applications do not automatically register new schemas.
+Register schemas outside of the client application to control when schemas are registered with |sr-long| and how they evolve.
+
+Within the application, disable automatic schema registration by setting the configuration parameter `auto.register.schemas=false`, as shown in the examples below.
+
+.. sourcecode:: java
+
+   props.put(AbstractKafkaAvroSerDeConfig.AUTO_REGISTER_SCHEMAS, false);
+
+To manually register the schema outside of the application, send the schema to |sr| and associate it with a subject, in this case `transactions-value`.  It returns a schema id of `1`.
```

Comment:
`the schema` -> `a schema` ?

It's unclear if the subject must be `transaction-value` or if user could pick any name?